File: test-suite/src/test/java/io/micronaut/docs/basics/BookController.java
package io.micronaut.docs.basics;

import io.micronaut.http.HttpStatus;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Post;
import io.micronaut.http.annotation.Status;

@Controller("/amazon")
public class BookController {

    @Post(value = "/book/{title}", consumes = {MediaType.APPLICATION_JSON, MediaType.APPLICATION_FORM_URLENCODED})
    @Status(HttpStatus.CREATED)
    Book save(String title) {
        return new Book(title);
    }
}

File: test-suite-python/src/test/python/micronaut/docs/basics/BookController.py
from micronaut.http import HttpStatus, MediaType
from micronaut.http.annotation import Controller, Post, Status

from .Book import Book


@Controller("/amazon")
class BookController:

    @Status(HttpStatus.CREATED)
    @Post(
        value="/book/{title}",
        consumes=[MediaType.APPLICATION_JSON, MediaType.APPLICATION_FORM_URLENCODED],
    )
    def save(self, title: str) -> Book:
        return Book(title)
